src/job.js

```diff
@@ -35,6 +35,8 @@ export default class Job extends EventEmitter {
 
   executeOn(thread) {
     thread
+      .off('progress')
+      .on('progress', (e) => {this.emit('progress', e)} )
```

Comment:
Should use either arrow function or `.bind()` consistently here across all `.on()`/`.once()`